stan/math/opencl/opencl_context.hpp

```diff
@@ -139,10 +140,10 @@ class opencl_context_base {
   cl::Context context_;  // Manages the the device, queue, platform, memory, etc
   cl::CommandQueue command_queue_;       // job queue for device, one per device
   std::vector<cl::Platform> platforms_;  // Vector of available platforms
-  cl::Platform platform_;                // The platform for compiling kernels
+  std::vector<cl::Platform> platform_;   // The platform for compiling kernels
   std::string platform_name_;  // The platform such as NVIDIA OpenCL or AMD SDK
+  std::vector<cl::Device> device_;   // The selected OpenCL device
```

Comment:
Why are these two now vectors?

Reply:
This looks odd but if you look at the accessor below we actually treat this like a vector already, but we'd essentially make a vector each time we call `device()`

https://github.com/stan-dev/math/pull/1804/files#diff-b87cced6b3cfba6bae2714b78737a40cR379

This whole thing was essentially the first thing I ever wrote in Stan lol